lib/coins/validators/index.js

```diff
@@ -61,6 +62,28 @@ function bech32Validator (network, address, opts) {
   return false
 }
 
+
 function isBech32Address (address, opts) {
   return bech32Validator('main', address, opts) || bech32Validator('test', address, opts)
 }
+
+function zecBech32Validator (network, address, opts) {
+  let decoded
+  try {
+    decoded = bech32.decode(address)
+  } catch (error) {
+    console.log('Failed to decode bech32 address')
+    return false
+  }
+
+  const data = bech32.fromWords(decoded.words)	
+  if (data.length !== 43) {	
+    console.log(`Invalid bech32 address length: ${data.length}`)	
+    return false	
+  }
+
+  if (network === 'main' && decoded.prefix === opts.mainNetPrefix) return true
+  if (network === 'test' && decoded.prefix === opts.testNetPrefix) return true
+  return false
+}
```

Comment:
This function has a lot of code equal to the `bech32Validator()` function. Consider merging this function with `bech32Validator()` and keep it all under the `bech32Validator` name

Reply:
I agree that there's some repeated code here, but it's doing a different thing in the end. If the witness version test and lenght are different a function with a "flag" would be harder to read.